### PR TITLE
Fix crash due to dangling pointer access in developer mode.

### PR DESCRIPTION
**Pull request by daschuer**
This fixes a regression #13885 since https://github.com/mixxxdj/mixxx/pull/12094

### CODE REVIEW

File: src/mixxxapplication.cpp [hunk 1]
```diff
@@ -140,7 +140,7 @@ void MixxxApplication::registerMetaTypes() {
     qRegisterMetaType<mixxx::FileInfo>("mixxx::FileInfo");
 }
 
-bool MixxxApplication::notify(QObject* target, QEvent* event) {
+bool MixxxApplication::notify(QObject* pTarget, QEvent* event) {
```
Comment (ronso0) [inline src/mixxxapplication.cpp:143]: what about pEvent, if you already touch this?

**Comment by daschuer** [comment]
Done

**Review by Swiftb0y — APPROVED** [review]
LGTM. Thank you.

**Comment by JoergAtGithub** [comment]
@daschuer Thank you for jumping in! The fix LGTM and I just tested it on Win11 -> Works as expected!

**Comment by ronso0** [comment]
Thanks @daschuer for quick fix!
Can confirm it doesn't crash anymore.

However, I want to note that I saw the crash for other event types, too, eg. `QChildEvent(ChildRemoved, ..)`.
So, presuming 
* #12094 is for diagnosis only
* we can't be sure it happens only for `QEvent::DeferredDelete`
* a crash can cause losing data from the current session
* users may run Mixxx with --developer in regular session to have MIDI Through for MIDI4Light

shall we tie to another developer option, eg. a certain log level?

**Comment by JoergAtGithub** [comment]
> However, I want to note that I saw the crash for other event types, too, eg. `QChildEvent(ChildRemoved, ..)`.

Than I see only two options:

1. Cache the object informantion before calling notify, which will affect the general performance in developer mode
2. Remove the printout of the target object information and just print the information about the event type

**Comment by ronso0** [comment]
3. implementing #8356 would "fix" it IMO since this eliminates the only reason I know to run Mixxx with --developer in regular sessions

**Pull request merged**